packages/core/test/validate_test.js

```diff
@@ -35,6 +35,20 @@ describe("Validation", () => {
 
       expect(isValid(schema, { foo: "bar" })).to.be.false;
     });
+
+    it("should allow extra ui fields", () => {
```

Comment:
Can you add some tests which involve using the `ui` property in more deeply nested examples, with references, etc.?

Reply:
> Thanks for making this! I wonder if it might be better to make the `ui` property as below so that it would be easier to transition old uiSchemas to bring them to the new format:
> 
> ```
> ui: {
>     "ui:widget": "x",
> }
> ```

I am not sure about this, we could do and it would be a case of just changing the merge in SchemaField but as it is changing to under ui field having extra ui: item is extra typing.

> Also, perhaps we can support `uiSchema` (in addition to the `ui` property) in v2, but just deprecate it, and remove it in a later version. We could make it so that anything in the `ui` property overrides anything that was already in the `uiSchema`.

I believe this is the case with this functionality.

I will do something in the playground here and add some additional tests allthough this is working for me with some complex schemas under my if/then/else stuff